src/components/Table/TableHead/TableHeader.js

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable react/button-has-type */
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { settings } from 'carbon-components';
+import { ArrowUp20 as Arrow, ArrowsVertical20 as Arrows } from '@carbon/icons-react';
+
+const sortStates = {
+  NONE: 'NONE',
+  DESC: 'DESC',
+  ASC: 'ASC',
+};
+
+const { prefix } = settings;
+
+const translationKeys = {
+  iconDescription: 'carbon.table.header.icon.description',
+};
+
+const translateWithId = (key, { sortDirection, isSortHeader }) => {
+  if (key === translationKeys.iconDescription) {
+    if (isSortHeader) {
+      // When transitioning, we know that the sequence of states is as follows:
+      // NONE -> ASC -> DESC -> NONE
+      if (sortDirection === sortStates.NONE) {
+        return 'Sort rows by this header in ascending order';
```

Comment:
this needs to be translated, right?

Reply:
I don't know,  I think that @sunilkotiyaibm copied this file from carbon to temporarily add a forwarding ref.

Reply:
These ref changes need to be put into a PR on the Carbon repo still. I will open an issue to track.

Reply:
> These ref changes need to be put into a PR on the Carbon repo still. I will open an issue to track.

[Issue](https://github.com/IBM/carbon-addons-iot-react/issues/862) created